src/Facets/EcoFacet.sol

```diff
@@ -0,0 +1,264 @@
+                _bridgeData.minAmount -
+                _ecoData.solverReward;
+        }
+
+        _startBridge(_bridgeData, _ecoData);
+    }
+
+    /// Internal Methods ///
+
+    function _getEcoChainId(
+        uint256 _lifiChainId
+    ) private view returns (uint64) {
+        // Map LiFi Tron chain ID to Eco protocol Tron chain ID
+        if (_lifiChainId == LIFI_CHAIN_ID_TRON) {
+            return ECO_CHAIN_ID_TRON;
+        }
+        // Map LiFi Solana chain ID to Eco protocol Solana chain ID
+        if (_lifiChainId == LIFI_CHAIN_ID_SOLANA) {
+            return ECO_CHAIN_ID_SOLANA;
+        }
+
+        // For EVM chains, ensure the chain ID fits within uint64 bounds
+        // Most EVM chain IDs are well below this limit, but we check to be safe
+        if (_lifiChainId > type(uint64).max) {
+            revert InvalidConfig();
+        }
+
+        return uint64(_lifiChainId);
+    }
+
+    function _buildReward(
+        ILiFi.BridgeData memory _bridgeData,
+        EcoData calldata _ecoData,
+        bool isNative,
+        uint256 totalAmount
+    ) private view returns (IEcoPortal.Reward memory) {
+        IEcoPortal.TokenAmount[] memory rewardTokens;
+        if (!isNative) {
+            rewardTokens = new IEcoPortal.TokenAmount[](1);
+            rewardTokens[0] = IEcoPortal.TokenAmount({
+                token: _bridgeData.sendingAssetId,
+                amount: totalAmount
+            });
+        } else {
+            rewardTokens = new IEcoPortal.TokenAmount[](0);
+        }
+
+        return
+            IEcoPortal.Reward({
+                creator: msg.sender,
+                prover: _ecoData.prover,
+                deadline: _ecoData.rewardDeadline,
+                nativeAmount: isNative ? totalAmount : 0,
+                tokens: rewardTokens
+            });
+    }
+
+    function _startBridge(
+        ILiFi.BridgeData memory _bridgeData,
+        EcoData calldata _ecoData
+    ) internal {
+        // Calculate values once
+        bool isNative = LibAsset.isNativeAsset(_bridgeData.sendingAssetId);
+        uint256 totalAmount = _bridgeData.minAmount + _ecoData.solverReward;
+
+        // Build reward for the intent
+        IEcoPortal.Reward memory reward = _buildReward(
+            _bridgeData,
+            _ecoData,
+            isNative,
+            totalAmount
+        );
+
+        // Get the destination chain ID in Eco format
+        uint64 destination = _getEcoChainId(_bridgeData.destinationChainId);
+
+        // Prepare token approval if needed for ERC20 tokens
+        if (!isNative) {
+            LibAsset.maxApproveERC20(
+                IERC20(_bridgeData.sendingAssetId),
+                address(PORTAL),
+                totalAmount
+            );
+        }
+
+        // Publish and fund the intent with encoded route
+        PORTAL.publishAndFund{ value: isNative ? totalAmount : 0 }(
+            destination,
+            _ecoData.encodedRoute,
```

Comment:
So I guess that the amount which will be bridged (aka intent input amount) will be somewhere encoded in encodedRoute (since we are not passing any other amount besides the reward), correct?

So what happens with positive slippage in case of a pre-bridge swap? We will update the minAmount (which in case of native sends more funds but in case of ERC20 is not being used).  
So I feel that the ERC20 positive slippage could get lost in the diamond, is that correct?

I think we should be very aware of this and document this limitation of the facet in the facet itself, if that is indeed the case. If someone bridges a large amount, the positive slippage can be significant.

Reply:
fixed: 19147bbfa8d02cae69c23960e3241ad05c606b7b